### ISSUE TITLE
Delayed explosion of inner flame

### ISSUE THREAD

**Issue by dillytin** [crawl/crawl#2036]
Version: 0.27-a0-1555-g0e7b3fb1d2

![Animation](https://user-images.githubusercontent.com/65288602/126041528-5aa6012d-daee-4de5-be60-b3cd6f3c5e8c.gif)
(just ignore some weird tiles like flame cloud or player tile)

The inner-flamed two-headed ogre died and then exploded.

**Comment by ebering** [comment]
this animated gif is surprisingly hard to follow. a step by step description of what happened would really help in trying to replicate this.

**Comment by ebering** [comment]
ok I've replicated it and found a fix. if you squint real quick you see that the ogre is poisoned and dying of poison

**Comment by PleasingFungus** [comment]
I liked the gif :)
